<a href="https://colab.research.google.com/github/Claudiap9/QualityRedWine/blob/main/BDI_475_Final_Project_Claudia_Park.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# BDI 475 Final Project: Red Wine Quality

For this project I will be analyzing what determines the "Quality" of red wine.

Input variables (based on physicochemical tests):

1 - fixed acidity
Majority of the acid contained in wine, which also does not evaporate

2 - volatile acidity
Acetic acid in wine (high levels can lead to unpleasant vinigar flavor)

3 - citric acid
Adds freshness to the flavor (found in small amounts)

4 - residual sugar
Remaining sugar after fermentation

5 - chlorides
Amount of salt

6 - free sulfur dioxide
free form of SO2 (prevents microbial growth and the oxidation of wine)

7 - total sulfur dioxide
free and bound forms of SO2

8 - density
the density of wine compared to water (depends on alchohol and sugar levels)

9 - pH
Describe how acidic the wine is on a scale of 0 to 14

10 - sulphates
wine additive which can contribute to sulfur dioxide gas levels (acts as an antimicrobial and antioxidant)

11 - alcohol
alcohol levels by percentage

Output variable (based on sensory data):

12 - quality (score between 0 and 10)

In [1]:
import pandas as pd
import numpy as np
import plotly.graph_objects as go
import plotly.express as px

In [2]:
df_wine = pd.read_csv('https://github.com/Claudiap9/QualityRedWine/raw/main/winequality-red.csv')

In [3]:
df_wine

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
0               7.4             0.700         0.00  ...       0.56      9.4        5
1               7.8             0.880         0.00  ...       0.68      9.8        5
2               7.8             0.760         0.04  ...       0.65      9.8        5
3              11.2             0.280         0.56  ...       0.58      9.8        6
4               7.4             0.700         0.00  ...       0.56      9.4        5
...             ...               ...          ...  ...        ...      ...      ...
1594            6.2             0.600         0.08  ...       0.58     10.5        5
1595            5.9             0.550         0.10  ...       0.76     11.2        6
1596            6.3             0.510         0.13  ...       0.75     11.0        6
1597            5.9             0.645         0.12  ...       0.71     10.2        5
1598            6.0             0.310         0.47  ...       0.66     11.0        6

[1599 rows x 12 columns]

In [4]:
df_wine.shape

(1599, 12)

In [5]:
df_wine.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1599 entries, 0 to 1598
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         1599 non-null   float64
 1   volatile acidity      1599 non-null   float64
 2   citric acid           1599 non-null   float64
 3   residual sugar        1599 non-null   float64
 4   chlorides             1599 non-null   float64
 5   free sulfur dioxide   1599 non-null   float64
 6   total sulfur dioxide  1599 non-null   float64
 7   density               1599 non-null   float64
 8   pH                    1599 non-null   float64
 9   sulphates             1599 non-null   float64
 10  alcohol               1599 non-null   float64
 11  quality               1599 non-null   int64  
dtypes: float64(11), int64(1)
memory usage: 150.0 KB


In [6]:
df_wine.head()

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
0            7.4              0.70         0.00  ...       0.56      9.4        5
1            7.8              0.88         0.00  ...       0.68      9.8        5
2            7.8              0.76         0.04  ...       0.65      9.8        5
3           11.2              0.28         0.56  ...       0.58      9.8        6
4            7.4              0.70         0.00  ...       0.56      9.4        5

[5 rows x 12 columns]

In [7]:
df_wine.tail()

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
1594            6.2             0.600         0.08  ...       0.58     10.5        5
1595            5.9             0.550         0.10  ...       0.76     11.2        6
1596            6.3             0.510         0.13  ...       0.75     11.0        6
1597            5.9             0.645         0.12  ...       0.71     10.2        5
1598            6.0             0.310         0.47  ...       0.66     11.0        6

[5 rows x 12 columns]

In [8]:
df_wine.sample(950)

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
822             6.7              0.54         0.13  ...       0.64      9.8        5
1412            8.2              0.24         0.34  ...       0.94     10.9        6
1049            9.3              0.50         0.36  ...       0.77     10.8        6
938             7.2              0.38         0.38  ...       0.72     12.9        7
935             8.8              0.30         0.38  ...       0.72     11.8        6
...             ...               ...          ...  ...        ...      ...      ...
862             7.5              0.42         0.32  ...       0.44     10.4        5
1532            7.2              0.53         0.13  ...       0.68      9.9        6
341            10.6              0.42         0.48  ...       0.87     11.3        6
1273            7.5              0.58         0.20  ...       0.43      9.3        5
867             6.9              0.51         0.23  ...       0.84     11.2        6

[950 rows x 12 columns]

In [9]:
df_wine.describe()

fixed acidity  volatile acidity  ...      alcohol      quality
count    1599.000000       1599.000000  ...  1599.000000  1599.000000
mean        8.319637          0.527821  ...    10.422983     5.636023
std         1.741096          0.179060  ...     1.065668     0.807569
min         4.600000          0.120000  ...     8.400000     3.000000
25%         7.100000          0.390000  ...     9.500000     5.000000
50%         7.900000          0.520000  ...    10.200000     6.000000
75%         9.200000          0.640000  ...    11.100000     6.000000
max        15.900000          1.580000  ...    14.900000     8.000000

[8 rows x 12 columns]

The quality of wine in this data ranges from 3 to 8, with a mean of 5.64. 

In [10]:
fig_quality_distribution = px.box(df_wine,
                                  x = 'quality',
                                  title = 'Range of Quality Score')
fig_quality_distribution.show()

In [11]:
fig_correlation = px.density_heatmap(df_wine,
                                     facet_col_wrap = 12,
                                     title = 'Correlation between Variables')

fig_correlation.show()